fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1352,6 +1359,180 @@ void simpleCrossPartitionQuery() {
         }
     }
 
+    static Stream<Arguments> findStartingPartitionTest() {
+        return Stream.of(true, false).map(Arguments::of);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void findStartingPartitionTest(boolean isSynthetic) {
```

Comment:
It would be good to have some more functional tests that create records with duplicates, and non duplicates across many times, and then iterating over the entire time range to validate.

Reply:
Right. That's what I was looking into

Reply:
Update: done, and merged in a separate PR